provisioning/provisioning-service-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/service/configs/EnrollmentGroup.java

```diff
@@ -94,10 +96,22 @@
 @SuppressWarnings("unused") // A number of private members are unused but may be filled in or used by serialization
 public class EnrollmentGroup extends Serializable
 {
-    // the enrollment group identifier
     private static final String ENROLLMENT_GROUP_ID_TAG = "enrollmentGroupId";
+
+    /**
+     * The unique identifier for this enrollment group.
+     *
+     * <p>
+     * A valid enrollment group Id shall follow this criteria.
+     * A case-sensitive string (up to 128 char long)
+     * of ASCII 7-bit alphanumeric chars
+     * + {'-', ':', '.', '+', '%', '_', '#', '*', '?', '!', '(', ')', ',', '=', '@', ';', '$', '''}.
```

Comment:
how about we link to the ms doc instead, so that we don't have to update this if service changes their criteria? we don't do any client side validations, right?

Reply:
I like that idea, but this PR is really just moving existing documentation around, so I'll save editing it for later